src/client/core/utils/shared/visibility.ts

```diff
@@ -1,15 +1,26 @@
 import adapter from './adapter/index';
 import { isNotVisibleNode, hasDimensions } from './style';
 
+function hiddenUsingStyles (el: HTMLElement): boolean {
+    return adapter.style.get(el, 'visibility') === 'hidden' ||
+        adapter.style.get(el, 'display') === 'none';
+}
+
+function hiddenByRectangle (el: HTMLElement): boolean {
+    const elementRectangle = adapter.position.getElementRectangle(el);
+
+    if (elementRectangle.width === 0 || elementRectangle.height === 0)
+        return true;
+
+    return false;
```

Comment:
You can do it easier

```suggestion
    return elementRectangle.width === 0 || elementRectangle.height === 0;
```